frontend/catalyst/device/decomposition.py

```diff
@@ -282,17 +354,17 @@ def _get_probs(counts_outcome):
     return jax.numpy.array(prob_vector)
 
 
-def _get_expval(eigvals, prob_vector):
+def _expval_from_probs(eigvals, prob_vector):
     """From the observable eigenvalues and the probability vector
     it calculates the expectation value."""
     expval = jax.numpy.dot(jax.numpy.array(eigvals), prob_vector)
     return expval
 
 
-def _get_var(eigvals, prob_vector):
+def _var_from_probs(eigvals, prob_vector):
     """From the observable eigenvalues and the probability vector
     it calculates the variance."""
-    var = jax.numpy.dot(prob_vector, (eigvals**2)) - jax.numpy.dot(prob_vector, eigvals)
+    var = jax.numpy.dot(prob_vector, (eigvals**2)) - jax.numpy.dot(prob_vector, eigvals) ** 2
```

Comment:
This slipped under the radar because we were testing that things could run, and produced the right type and shape of result, but not the result was correct compared to a trusted device or an analytic result 🐛